Character.cpp

```diff
@@ -87,31 +88,54 @@ int Character::takeDamage(int damage)
 }
 
 
-#include <assert>
+//#include <assert>
 void Character::attackInternal(Character& other)
 {
     if( other.hitPoints <= 0 )
     {
         /*
         When you defeat another Character: 
-            a) your stats are restored to their initial value
+            a) your stats are restored to their initial value if they are lower than it.
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
-        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
+        //assert(false);
+        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;
+ 
+        if(hitPoints < *initialHitPoints)
+        {
+            hitPoints = *initialHitPoints;
+        }
+        
+        if (armor < *initialArmorLevel)
+        {
+            armor = *initialArmorLevel;
+        }
+
+        if (attackDamage < *initialAttackDamage)
+        {
+            attackDamage = *initialAttackDamage;
+        }  
+          
+        hitPoints *= 1.1;
+        armor *= 1.1;
+        attackDamage *= 1.1;
+
+        *initialHitPoints = hitPoints;
+        *initialArmorLevel = armor;
+        *initialAttackDamage = attackDamage; 
```

Comment:
Can you D.R.Y. this into a free function?  
You're doing the same action with each pair of variables 3x.
- [ ] D.R.Y. refactor with a free function